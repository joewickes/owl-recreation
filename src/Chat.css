* {
  margin: 0px;
  border: 0px;
}

.display-message {
  display: flex;
  align-items: center;
  align-self: flex-start;
}

.message-heading {
  display: flex;
}

.participant-name {
  margin-right: 10px;
}

.message-left {
  display: flex;
  align-items: center;
  justify-content: center;
  border: 1px solid black;
  border-radius: 50%;
  width: 40px;
  margin: 10px;
}

.message-left img {
  width: 100%;
  border-radius: 50%;
}
