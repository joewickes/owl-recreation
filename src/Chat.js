import React from 'react';
import './Chat.css';

class Chat extends React.Component {
  static defaultProps = {
    name: '',
    type: '',
    message: '',
    avatar: '',
    time: 0,
    timestamp: 0,
  };
  
  render() {
    const {name, type, message, avatar, time} = this.props;

    let displayText = null;

    const postTime = new Date(time);
    const hours = postTime.getHours();
    const minutes = postTime.getMinutes();

    if (type === 'join') {
      // Name joined
      displayText = <p className="display-text"><b>{name}</b> joined</p>;
    } else if (type === 'leave') {
      // Name left
      displayText = <p className="display-text"><b>{name}</b> left</p>;
    } else if (type === 'join-stage') {
      // Name joined the stage
      displayText = <p className="display-text"><b>{name}</b> joined the stage</p>;
    } else if (type === 'leave-stage') {
      // Name left the stage
      displayText = <p className="display-text"><b>{name}</b> left the stage</p>;
    } else if (type === 'clap') {
      // Name clapped
      displayText = <p className="display-text"><b>{name}</b> clapped</p>;
    } else if (type === 'raise-hand') {
      // Name raised their hand
      displayText = <p className="display-text"><b>{name}</b> raised their hand</p>;
    } else if (type === 'thumbs-up') {
      // Name gave a thumbs up
      displayText = <p className="display-text"><b>{name}</b> gave a thumbs up</p>;
    } else if (type === 'thumbs-down') {
      // Name gave a thumbs down
      displayText = <p className="display-text"><b>{name}</b> gave a thumbs down</p>;
    } else if (type === 'message') {
      // Icon, name, time?, and message under name
      displayText = (
        <div className="display-message">
          <div className="message-left">
            <img src={avatar} alt={`${name}'s Avatar`} />
          </div>
          <div className="message-right">
            <div className="message-heading">
              <p className="participant-name">{name}</p>
              <p>{`${hours}:${minutes}`}</p>
            </div>
            <p>{message}</p>
          </div>
        </div> 
      );
    }

    return (
      <div className="Chat">
        {displayText}
      </div>
    );
  }
}

export default Chat;