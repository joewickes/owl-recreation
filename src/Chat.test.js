import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Chat from './Chat';

describe('Chat Component', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Chat />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders a message', () => {
    const tree = renderer
      .create(<Chat name="Bob" type="message" message="Hey , I'm here" avatar="https://robohash.org/itaquevoluptatumrerum.jpg?size=200x200&set=set1" time={1548852646559} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders a join', () => {
    const tree = renderer
      .create(<Chat name="Sally" type="join" />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders a leave', () => {
    const tree = renderer
      .create(<Chat name="Sally" type="leave" />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders a join stage', () => {
    const tree = renderer
      .create(<Chat name="Sally" type="join-stage" />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders a leave stage', () => {
    const tree = renderer
      .create(<Chat name="Sally" type="leave-stage" />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders a thumbs up', () => {
    const tree = renderer
      .create(<Chat name="Sally" type="thumbs-up" />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders a thumbs down', () => {
    const tree = renderer
      .create(<Chat name="Sally" type="thumbs-down" />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders a raised hand', () => {
    const tree = renderer
      .create(<Chat name="Sally" type="raised-hand" />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders a clap', () => {
    const tree = renderer
      .create(<Chat name="Sally" type="clap" />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});