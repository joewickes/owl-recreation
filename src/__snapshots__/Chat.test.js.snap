// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Chat Component renders a clap 1`] = `
<div
  className="Chat"
>
  <p
    className="display-text"
  >
    Sally
     clapped
  </p>
</div>
`;

exports[`Chat Component renders a join 1`] = `
<div
  className="Chat"
>
  <p
    className="display-text"
  >
    Sally
     joined
  </p>
</div>
`;

exports[`Chat Component renders a join stage 1`] = `
<div
  className="Chat"
>
  <p
    className="display-text"
  >
    Sally
     joined the stage
  </p>
</div>
`;

exports[`Chat Component renders a leave 1`] = `
<div
  className="Chat"
>
  <p
    className="display-text"
  >
    Sally
     left
  </p>
</div>
`;

exports[`Chat Component renders a leave stage 1`] = `
<div
  className="Chat"
>
  <p
    className="display-text"
  >
    Sally
     left the stage
  </p>
</div>
`;

exports[`Chat Component renders a message 1`] = `
<div
  className="Chat"
>
  <div
    className="display-message"
  >
    <div
      className="message-left"
    >
      <img
        alt="Bob's Avatar"
        src="https://robohash.org/itaquevoluptatumrerum.jpg?size=200x200&set=set1"
      />
    </div>
    <div
      className="message-right"
    >
      <div
        className="message-heading"
      >
        <p
          className="participant-name"
        >
          Bob
        </p>
        <p>
          7:50
        </p>
      </div>
      <p>
        Hey , I'm here
      </p>
    </div>
  </div>
</div>
`;

exports[`Chat Component renders a raised hand 1`] = `
<div
  className="Chat"
/>
`;

exports[`Chat Component renders a thumbs down 1`] = `
<div
  className="Chat"
>
  <p
    className="display-text"
  >
    Sally
     gave a thumbs down
  </p>
</div>
`;

exports[`Chat Component renders a thumbs up 1`] = `
<div
  className="Chat"
>
  <p
    className="display-text"
  >
    Sally
     gave a thumbs up
  </p>
</div>
`;
